api-service/service/databases.go

```diff
@@ -432,26 +432,69 @@ func (as *APIService) GetDatabasesUsedLicensesPerHost(filter dto.GlobalFilter) (
 		return nil, err
 	}
 
+	clusters, err := as.Database.GetClusters(filter)
+	if err != nil {
+		return nil, err
+	}
+
+	clusterByHostnames := make(map[string]*dto.Cluster)
+	for i := range clusters {
+		for j := range clusters[i].VMs {
+			clusterByHostnames[clusters[i].VMs[j].Hostname] = &clusters[i]
+		}
+	}
+
+	m := make(map[string]map[string]dto.DatabaseUsedLicensePerHost)
+
 	var licensesPerHost []dto.DatabaseUsedLicensePerHost
 
 licenses:
 	for _, v := range licenses {
+
 		for i, v2 := range licensesPerHost {
 			if v.Hostname == v2.Hostname && v.LicenseTypeID == v2.LicenseTypeID {
 				licensesPerHost[i].DatabaseNames = append(licensesPerHost[i].DatabaseNames, v.DbName)
 				continue licenses
 			}
 		}
 
-		licensesPerHost = append(licensesPerHost, dto.DatabaseUsedLicensePerHost{
-			Hostname:        v.Hostname,
-			DatabaseNames:   []string{v.DbName},
-			LicenseTypeID:   v.LicenseTypeID,
-			Description:     v.Description,
-			Metric:          v.Metric,
-			UsedLicenses:    v.UsedLicenses,
-			ClusterLicenses: v.ClusterLicenses,
-		})
+		c, found := clusterByHostnames[v.Hostname]
+		if found {
+			clusterLicenses, ok := m[c.Name]
+			if !ok {
+				clusterLicenses = make(map[string]dto.DatabaseUsedLicensePerHost)
+				m[c.Name] = clusterLicenses
+			}
+
+			ll, ok := clusterLicenses[v.LicenseTypeID]
+			if !ok {
+				ll = dto.DatabaseUsedLicensePerHost{
```

Comment:
Since everything is in common of this "ll" variable besides the `ClusterLicenses`, why don't we merge the common part so that it's more clear that the difference here is only the `ClusterLicenses` value?